Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -345,8 +345,11 @@ public void startPausing() {
 
   @Override
   public void runToPosition(@NotNull XSourcePosition position, @Nullable XSuspendContext context) {
-    // todo implement
-    resume(context);
+    if (myLatestCurrentIsolateId != null && mySuspendedIsolateIds.contains(myLatestCurrentIsolateId)) {
+      // Set a temporary breakpoint and resume.
+      myVmServiceWrapper.addTemporaryBreakpoint(position, myLatestCurrentIsolateId);
+      myVmServiceWrapper.resumeIsolate(myLatestCurrentIsolateId, null);
```

Comment:
I'm assuming that when the user does run-to-cursor, they're expecting the current isolate to resume, and break when it reaches the current cursor position. Other isolates may be paused debugging completely separate bits of code and it might be unexpected to resume them.